op_crates/fetch/lib.rs

```diff
@@ -212,11 +216,20 @@ pub async fn op_fetch_read(
   let mut response = RcRef::map(&resource, |r| &r.response).borrow_mut().await;
   let cancel = RcRef::map(resource, |r| &r.cancel);
   let maybe_chunk = response.chunk().or_cancel(cancel).await??;
+  //TODO(Soremwar)
+  //Reqwest must have an efficient way to access bytes of the chunk
+  //This will do for now
   if let Some(chunk) = maybe_chunk {
-    // TODO(ry) This is terribly inefficient. Make this zero-copy.
-    Ok(json!({ "chunk": &*chunk }))
+    let bytes = chunk.len();
+    if (start + length) > bytes {
+      (&mut zero_copy[0])[0..bytes - start]
+        .copy_from_slice(&chunk[start..bytes]);
+    } else {
+      (&mut zero_copy[0]).copy_from_slice(&chunk[start..start + length]);
+    }
+    Ok(json!({ "bytes": bytes as u64 }))
   } else {
-    Ok(json!({ "chunk": null }))
+    Ok(json!({"bytes": 0}))
```

Comment:
@Soremwar I think you want to lift code for `HttpBody` from https://github.com/denoland/deno/pull/8781 
I removed it a few days ago cause I couldn't find where it was used, now I remember that it was used by previous implementation of `fetch` that used zero copy.